### PR TITLE
Unification and clearing up common code for all extensions

### PR DESCRIPTION
**Pull request by sawradip**

### CODE REVIEW

File: autogen/composio_autogen/autogen_toolspec.py [hunk 1]
```diff
@@ -1,100 +1,16 @@
...
-fallback_values = {
-    'string': "",
-    'number': 0.0,
-    'integer': 0.0,
-    'boolean': False,
-    'object': {},
-    'array': []
-}
-
-def pydantic_model_from_param_schema(param_schema):
```
Comment (kaavee315) [inline autogen/composio_autogen/autogen_toolspec.py:30]: This was not being used?

Comment (sawradip) [reply autogen/composio_autogen/autogen_toolspec.py:30]: Being used, but moved all these too the core, as shared.

File: core/composio/sdk/shared_utils.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        param_type = param_schema['type']
+        param_title = param_schema['title'].replace(" ", "")
+
+        if param_type in schema_type_python_type_dict:
+            signature_param_type = schema_type_python_type_dict[param_type]
+        else:
+            signature_param_type = pydantic_model_from_param_schema(param_schema)
+
+        param_default = param_schema.get('default', fallback_values[param_type])
+        param_annotation = Annotated[signature_param_type, param_schema.get('description',
```
Comment (kaavee315) [inline core/composio/sdk/shared_utils.py:191]: Changed this function in master. Please have a look and update

**Review by kaavee315 — APPROVED** [review]

File: core/composio/sdk/shared_utils.py (hunk 2 above)
Comment (utkarsh-dixit) [reply core/composio/sdk/shared_utils.py:191]: @sawradip is this resolved?

**Review by utkarsh-dixit — APPROVED** [review]

File: core/composio/sdk/shared_utils.py (hunk 2 above)
Comment (kaavee315) [reply core/composio/sdk/shared_utils.py:191]: Yes, it is not resolved. Could you check @sawradip

Comment (sawradip) [reply core/composio/sdk/shared_utils.py]: Fixed.

File: core/composio/sdk/shared_utils.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        else:
+            optional_fields[prop_name] = (signature_prop_type, 
+                                Field(title=prop_title, 
+                                    default=prop_default
+                                    ))
+    fieldModel = create_model(param_title, **required_fields, **optional_fields)
+    return fieldModel
+        
+        
+# def get_signature_format_from_schema_params(
```
Comment (kaavee315) [inline core/composio/sdk/shared_utils.py:164]: Can we remove this?

**Review by kaavee315 — APPROVED** [review]

**Pull request merged**